lgc/patch/Patch.cpp

```diff
@@ -219,9 +222,22 @@ void Patch::addPasses(PipelineState *pipelineState, legacy::PassManager &passMgr
 //
 // @param [in/out] passMgr : Pass manager to add passes to
 void Patch::addOptimizationPasses(legacy::PassManager &passMgr) {
+  // Use CodeGenOpt::Aggressive (i.e. 3) by default as this is what we have been
+  // doing so far.
+  CodeGenOpt::Level OLvl = CodeGenOpt::Aggressive;
+  switch (cl::OptLevel) {
+  default:
+    WithColor::error(errs(), "llpc") << "invalid optimization level.\n";
+    return;
+  case ' ': break;
+  case '0': OLvl = CodeGenOpt::None; break;
+  case '2': OLvl = CodeGenOpt::Aggressive; break;
+  case 'c': OLvl = CodeGenOpt::None; break; // -O0 for now
+  }
+
   // Set up standard optimization passes.
   if (!cl::UseLlvmOpt) {
-    unsigned optLevel = 3;
+    unsigned optLevel = OLvl;
```

Comment:
We can remove this local variable and use `OLvl` in `passMgr.add(createSimpleLoopUnrollPass(optLevel))`